CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -303,12 +306,14 @@ private static void BlocksToHtmlInner(HtmlTextWriter writer, Block block, Common
                             writer.WriteConstant("<h" + x.ToString(CultureInfo.InvariantCulture));
                             PrintPosition(writer, block);
                             writer.Write('>');
+                            WriteGithubHeaderIds(writer, block.InlineContent, settings, inlineStack);
```

Comment:
Store the `settings.AdditionalFeatures & CommonMarkAdditionalFeatures.GithubStyleHeaderIds` result in the start of BlocksToHtmlInner and check the condition before invoking the method.